### PR TITLE
Fix heating rate error

### PR DESCRIPTION
**Pull request by Crash-Evans**

### CODE REVIEW

File: blueprints/automation/tado_temp_offset.yaml [hunk 1]
```diff
@@ -43,35 +45,30 @@ blueprint:
...
-  heating_rate: >
-    {% set entities = device_entities(device_id("climate.schlafzimmer") ) %}
-    {% for s in entities %}
-      {% if 'heating' in s %}
-        {{ states(s) }}
-      {% endif %}
-    {% endfor %}
+  actual_temp: "{{ states(source_temp_sensor) | float }}"
+  run_flag: '{{ night_switch is not none and is_state(night_switch, "off") }}'
+  heating_rate: '{{ state_attr(target_tado, "heating_rate") | float }}'  # Adjust based on Tado's attributes
```
Comment (Clemensao) [inline blueprints/automation/tado_temp_offset.yaml]: This won't work in my configuration. heating_rate is no attribute of the climate entity but a seperate sensor-entity connected to the device. By default, the tado integration creates this sensor with an entity_id ending on "heating".

I suggest this approach:
`heating_rate: '{{ expand(device_entities(device_id(target_tado))) | selectattr("entity_id","match", "^sensor.*heating$") | map(attribute="state") | first | float }}'`

File: blueprints/automation/tado_temp_offset.yaml [hunk 2]
```diff
@@ -43,35 +45,30 @@ blueprint:
...
-  run_flag: '{{ night_switch == none or (night_switch != none and is_state(night_switch, "off")) }}'
-  heating_rate: >
-    {% set entities = device_entities(device_id("climate.schlafzimmer") ) %}
-    {% for s in entities %}
-      {% if 'heating' in s %}
-        {{ states(s) }}
-      {% endif %}
-    {% endfor %}
+  actual_temp: "{{ states(source_temp_sensor) | float }}"
+  run_flag: '{{ night_switch is not none and is_state(night_switch, "off") }}'
```
Comment (Clemensao) [inline blueprints/automation/tado_temp_offset.yaml]: What will happen if night_switch is not set? Nighttime is an optional input! 
I admit "run_flag" is not a good variable name. It should be s.th. like day_flag

`run_flag: '{{ night_switch is none or (night_switch is not none and is_state(night_switch, "off")) }}' # flag for determining the daytime-mode`

File: blueprints/automation/tado_temp_offset.yaml [hunk 3]
```diff
@@ -95,29 +93,38 @@ condition:
...
-      - condition: template
-        value_template: '{{ heating_rate|float <= heating_threshold|float }}'
-      - condition: template
-        value_template: '{{ ((run_flag == true and trigger.id == "day") or trigger.id == "night") or trigger_sensor == true }}'
-      - condition: template
-        value_template: "{{ (trigger.platform in ['state','event'] and trigger_sensor == true) or trigger_sensor == false  }}"
+        value_template: "{{ offset != 0 and actual_temp != 0 and heating_rate <= heating_threshold }}"
+      - condition: or
+        conditions:
+          - condition: template
```
Comment (Clemensao) [inline blueprints/automation/tado_temp_offset.yaml:97]: I'd rather delete this condition (line 99 + 100)

**Review by Clemensao — CHANGES_REQUESTED** [review]
Good work! Thanks for fixing my sloppy template :)
Please see my comments...

Cheers!

File: blueprints/automation/tado_temp_offset.yaml (hunk 1 above)
Comment (Crash-Evans) [reply blueprints/automation/tado_temp_offset.yaml]: I'm really new to this so it's great to learn! Agree as it threw up errors for me.

File: blueprints/automation/tado_temp_offset.yaml (hunk 2 above)
Comment (Crash-Evans) [reply blueprints/automation/tado_temp_offset.yaml]: I'll have a play about and see what it the debug looks like

**Comment by Crash-Evans** [comment]
I'll have a look at this over the next couple of days and see what comes back

**Pull request closed without merging**